base/sparse/sparsematrix.jl

```diff
@@ -431,6 +431,37 @@ end
 
 ## Unary arithmetic and boolean operators
 
+macro _unary_op(Ti,Tv,A,op)
```

Comment:
I think this macro could be called `unary_op_nz2z_z2z`, since it may map nonzeros to zero, but zeros remain zero. Could you make the argument order `(op, A, Tv, Ti)`?